### PR TITLE
434 fixed

### PR DESCRIPTION
**Pull request by brijeshnec**

### CODE REVIEW

**Review by amitrnec — APPROVED** [review]

**Review by amitrnec — APPROVED** [review]

File: HistoryQueryManager/src/main/java/eu/neclab/ngsildbroker/historyquerymanager/service/HistoryQueryService.java [hunk 1]
```diff
@@ -356,54 +373,106 @@ private Map<RemoteHost, Set<String>> getRemoteHostsForRetrieve(String tenant, St
...
+
+		for (Map.Entry<String, String> entry : paramMap.entrySet()) {
+			if (!queryStringBuilder.isEmpty()) {
+				queryStringBuilder.append("&");
+			}
+			try {
+				String key = URLEncoder.encode(entry.getKey(), StandardCharsets.UTF_8);
+				String value = URLEncoder.encode(entry.getValue(), StandardCharsets.UTF_8);
+				queryStringBuilder.append(key).append("=").append(value);
+			} catch (Exception ignored) {}
```
Comment (ScorpioBroker) [inline HistoryQueryManager/src/main/java/eu/neclab/ngsildbroker/historyquerymanager/service/HistoryQueryService.java]: log the exception at  least. broad try catch with exception and no error handling is prone to bite us in the future.

File: HistoryQueryManager/src/main/java/eu/neclab/ngsildbroker/historyquerymanager/service/HistoryQueryService.java [hunk 2]
```diff
@@ -356,54 +373,106 @@ private Map<RemoteHost, Set<String>> getRemoteHostsForRetrieve(String tenant, St
...
 	public Uni<Void> handleRegistryChange(BaseRequest req) {
 		tenant2CId2RegEntries.remove(req.getTenant(), req.getId());
 		if (req.getRequestType() != AppConstants.DELETE_REQUEST) {
-			for (RegistrationEntry regEntry : RegistrationEntry.fromRegPayload(req.getPayload())) {
+			List<RegistrationEntry> lsReg = RegistrationEntry.fromRegPayload(req.getPayload());
+			int count=0;
+			for (RegistrationEntry regEntry : lsReg) {
 				if (regEntry.retrieveTemporal() || regEntry.queryTemporal()) {
-					tenant2CId2RegEntries.put(req.getTenant(), req.getId(), regEntry);
+					tenant2CId2RegEntries.put(req.getTenant(), req.getId()+count++, regEntry);
```
Comment (ScorpioBroker) [inline HistoryQueryManager/src/main/java/eu/neclab/ngsildbroker/historyquerymanager/service/HistoryQueryService.java]: please change this. this is supposed to overwrite the entries on changes. i know it was not properly working to begin with like this but please have a look at the entity service for instance and do it like this thanks

**Review by ScorpioBroker — CHANGES_REQUESTED** [review]

File: HistoryQueryManager/src/main/java/eu/neclab/ngsildbroker/historyquerymanager/repository/HistoryDAO.java [hunk 3]
```diff
@@ -117,7 +117,7 @@ public Uni<Table<String, String, RegistrationEntry>> getAllRegistries() {
 								while (it2.hasNext()) {
 									Row row = it2.next();
 									result.put(tenant, row.getString(1),
-											DBUtil.getRegistrationEntry(row, tenant, logger));
+											List.of(DBUtil.getRegistrationEntry(row, tenant, logger)));
```
Comment (ScorpioBroker) [inline HistoryQueryManager/src/main/java/eu/neclab/ngsildbroker/historyquerymanager/repository/HistoryDAO.java]: this overwrites previous entries with the new one and doesn't add to it
do something like this from entity manager
return Uni.combine().all().unis(unis).combinedWith(list -> {
							Table<String, String, List<RegistrationEntry>> result = HashBasedTable.create();
							for (Object obj : list) {
								Tuple2<String, RowSet<Row>> tuple = (Tuple2<String, RowSet<Row>>) obj;
								String tenant = tuple.getItem1();
								RowIterator<Row> it2 = tuple.getItem2().iterator();
								while (it2.hasNext()) {
									Row row = it2.next();
									List<RegistrationEntry> entries = result.get(tenant, row.getString(1));
									if (entries == null) {
										entries = Lists.newArrayList();
										result.put(tenant, row.getString(1), entries);
									}
									entries.add(DBUtil.getRegistrationEntry(row, tenant, logger));
								}
							}
							return result;
						});

**Review by ScorpioBroker — CHANGES_REQUESTED** [review]

File: HistoryQueryManager/src/main/java/eu/neclab/ngsildbroker/historyquerymanager/repository/HistoryDAO.java (hunk 3 above)
Comment (brijeshnec) [reply HistoryQueryManager/src/main/java/eu/neclab/ngsildbroker/historyquerymanager/repository/HistoryDAO.java]: done

**Review by ScorpioBroker — APPROVED** [review]
lgtm

**Pull request merged**